### DISCUSSION TITLE
Raid cooldown/cancel should have an option to preserve bad omen effect

### DISCUSSION THREAD

**Discussion by sandrwich** [PurpurMC/Purpur#1042] [Feature Requests]
Currently, bad omen is explicitly removed when the player triggers a raid whilst in cooldown. It would be great to have this configurable.

I also don't particularly like the fact that cancelling a raid event removes the bad omen effect, as this prevents implementing such a feature on the plugin side without minor hacks, so it can be two separate settings or only one for both.

[0099-Raid-cooldown-setting.patch](https://github.com/PurpurMC/Purpur/blob/146750b2ded20c71cd297cae429e1d3ec6e30db4/patches/server/0099-Raid-cooldown-setting.patch)
```patch
-                    if (!org.bukkit.craftbukkit.event.CraftEventFactory.callRaidTriggerEvent(raid, player)) {
+                    if ((level.purpurConfig.raidCooldownSeconds != 0 && playerCooldowns.containsKey(player.getUUID())) || !org.bukkit.craftbukkit.event.CraftEventFactory.callRaidTriggerEvent(raid, player)) { // Purpur
                         player.removeEffect(MobEffects.BAD_OMEN);
                         return null;
                     }
```

**Comment by BillyGalbreath** [comment]
I've created an issue ticket for this. I'd consider this a bug and not a feature request ;) https://github.com/PurpurMC/Purpur/issues/1043
